Permission denied when building using Conda on Windows

### What is the bug?

I've been building GDAL from source successfully using Conda, but after ~4 weeks break I'm getting errors on Windows when building with `master`:

``` ps1
CMake Error at C:/docs/gdal/swig/python/modify_cpp_files.cmake:72 (file):
    file failed to open for writing (Permission denied):

      C:/docs/gdal/build/swig/python/extensions//ogr_wrap.cpp


  The system cannot find the batch label specified - VCEnd
C:\Program Files\Microsoft Visual Studio\2022\Professional\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(237,5): error MSB8066: Custom build for 'C:\docs\gdal\build\CMakeFiles\8a08d127f28cf8f225c1d616d8c9d602\ogr_wrap.cpp.rule' exited with code 1. [C:\docs\gdal\build\swig\python\python_binding.vcxproj]
  CMake Error at C:/docs/gdal/swig/python/modify_cpp_files.cmake:72 (file):
    file failed to open for writing (Permission denied):

      C:/docs/gdal/build/swig/python/extensions//gdal_wrap.cpp


  The system cannot find the batch label specified - VCEnd
C:\Program Files\Microsoft Visual Studio\2022\Professional\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(237,5): error MSB8066: Custom build for 'C:\docs\gdal\build\CMakeFiles\8a08d127f28cf8f225c1d616d8c9d602\gdal_wrap.cpp.rule' exited with code 1. [C:\docs\gdal\build\swig\python\python_binding.vcxproj]
  Building Custom Rule C:/docs/gdal/apps/CMakeLists.txt
```

I am running in a PowerShell `administrator` shell, and there are no special permissions or read-only flags on the files mentioned.
I thought it could be related to CMake trying to write the same file at the same time in parallel, but the only recent relevant change seems to be #12500, and I tried rebuilding with this removed (and deleting the build folder first). 

The line causing the issue has been there 3 years: https://github.com/OSGeo/gdal/blob/e8ee31b9bd460045b70706fb4d7654046921fc5f/swig/python/modify_cpp_files.cmake#L72

The files `C:/docs/gdal/build/swig/python/extensions//ogr_wrap.cpp` and ` C:/docs/gdal/build/swig/python/extensions//gdal_wrap.cpp` exist on disk when the build stops (the double `//` could be significant?). 

### Steps to reproduce the issue

```ps1
md c:\docs
cd c:\docs

conda create --yes --name gdal-docs-py313 python=3.13
cmake --build . --config Release
conda activate gdal-docs
conda install --yes -c conda-forge pip doxygen cmake proj ca-certificates swig

git clone https://github.com/OSGeo/gdal/
pip install -r C:\docs\gdal\doc\requirements.txt

cd c:\docs\gdal
md build
cd c:\docs\gdal\build
cmake -DGDAL_BUILD_OPTIONAL_DRIVERS=OFF -DOGR_BUILD_OPTIONAL_DRIVERS=OFF ..
cmake --build . --target html --config Release
```


### Versions and provenance

- Windows 11
- conda 24.3.0
- Python 3.13.2 (also tested with Python 3.14.0rc2)
- gdal master (5/9/25)

### Additional context

_No response_

--- a/cmake/helpers/GdalSwigBindings.cmake
+++ b/cmake/helpers/GdalSwigBindings.cmake
@@ -14,7 +14,7 @@ find_package(SWIG REQUIRED)
 function(gdal_swig_bindings)
   set(_options)
   set(_oneValueArgs BINDING)
-  set(_multiValueArgs "ARGS;DEPENDS;OUTPUT")
+  set(_multiValueArgs "ARGS;DEPENDS")
   cmake_parse_arguments(_SWIG "${_options}" "${_oneValueArgs}" "${_multiValueArgs}" ${ARGN})
   file(MAKE_DIRECTORY ${PROJECT_BINARY_DIR}/swig/${_SWIG_BINDING}/extensions)
   set(SWIG_ARGS -Wall ${_SWIG_ARGS} -I${PROJECT_SOURCE_DIR}/swig/include -I${PROJECT_SOURCE_DIR}/swig/include/${_SWIG_BINDING})
@@ -23,7 +23,6 @@ function(gdal_swig_bindings)
           TARGET gdalconst
           BINDING ${_SWIG_BINDING}
           ARGS ${SWIG_ARGS}
-          OUTPUT ${_SWIG_OUTPUT}
           DEPENDS ${GDAL_SWIG_COMMON_INTERFACE_FILES}
           ${_SWIG_DEPENDS}
           ${PROJECT_SOURCE_DIR}/swig/include/${_SWIG_BINDING}/typemaps_${_SWIG_BINDING}.i
@@ -35,7 +34,6 @@ function(gdal_swig_bindings)
             TARGET ${tgt} CXX
             BINDING ${_SWIG_BINDING}
             ARGS ${SWIG_ARGS}
-            OUTPUT ${_SWIG_OUTPUT}
             DEPENDS ${GDAL_SWIG_COMMON_INTERFACE_FILES}
             ${_SWIG_DEPENDS}
             ${PROJECT_SOURCE_DIR}/swig/include/${_SWIG_BINDING}/typemaps_${_SWIG_BINDING}.i
@@ -49,7 +47,7 @@ endfunction()
 function(gdal_swig_binding_target)
   set(_options CXX)
   set(_oneValueArgs "TARGET;BINDING")
-  set(_multiValueArgs "ARGS;DEPENDS;OUTPUT")
+  set(_multiValueArgs "ARGS;DEPENDS")
   cmake_parse_arguments(_SWIG "${_options}" "${_oneValueArgs}" "${_multiValueArgs}" ${ARGN})
   if (_SWIG_CXX)
     set(_OUTPUT ${PROJECT_BINARY_DIR}/swig/${_SWIG_BINDING}/extensions/${_SWIG_TARGET}_wrap.cpp)
@@ -58,13 +56,22 @@ function(gdal_swig_binding_target)
   endif ()
   if ("${_SWIG_BINDING}" STREQUAL "python")
     set(BINDING_LANGUAGE_OUTPUT "${PROJECT_BINARY_DIR}/swig/${_SWIG_BINDING}/osgeo/${_SWIG_TARGET}.py")
-  endif()
-  add_custom_command(
+    add_custom_command(
           OUTPUT ${_OUTPUT} ${BINDING_LANGUAGE_OUTPUT}
           COMMAND ${SWIG_EXECUTABLE} ${_SWIG_ARGS} ${SWIG_DEFINES}
           $<$<BOOL:${_SWIG_CXX}>:-c++> -${_SWIG_BINDING}
           -o ${_OUTPUT}
           ${PROJECT_SOURCE_DIR}/swig/include/${_SWIG_TARGET}.i
-          DEPENDS ${_SWIG_DEPENDS})
+          COMMAND ${CMAKE_COMMAND} ${WERROR_DEV_FLAG} "-DFILE=${_OUTPUT}" -P ${PROJECT_SOURCE_DIR}/swig/python/modify_cpp_files.cmake
+          DEPENDS ${_SWIG_DEPENDS} ${PROJECT_SOURCE_DIR}/swig/python/modify_cpp_files.cmake)
+  else()
+    add_custom_command(
+              OUTPUT ${_OUTPUT}
+              COMMAND ${SWIG_EXECUTABLE} ${_SWIG_ARGS} ${SWIG_DEFINES}
+              $<$<BOOL:${_SWIG_CXX}>:-c++> -${_SWIG_BINDING}
+              -o ${_OUTPUT}
+              ${PROJECT_SOURCE_DIR}/swig/include/${_SWIG_TARGET}.i
+              DEPENDS ${_SWIG_DEPENDS})
+  endif()
   set_source_files_properties(${SWIG_OUTPUT} PROPERTIES GENERATED 1)
 endfunction()

--- a/swig/python/CMakeLists.txt
+++ b/swig/python/CMakeLists.txt
@@ -58,11 +58,7 @@ set(GDAL_PYTHON_CSOURCES
     -threads
     -relativeimport
     -outdir
-    ${CMAKE_CURRENT_BINARY_DIR}/osgeo
-    OUTPUT
-    ${GDAL_PYTHON_PYSOURCES}
-    ${GDAL_PYTHON_CSOURCES}
-    DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/modify_cpp_files.cmake)
+    ${CMAKE_CURRENT_BINARY_DIR}/osgeo)
 
   # gdal_array_wrap.cpp when NumPy exist
   if (Python_NumPy_FOUND)
@@ -75,6 +71,7 @@ set(GDAL_PYTHON_CSOURCES
         -I${PROJECT_SOURCE_DIR}/swig/include/python/docs ${SWIG_PY3_DEPRECATED} -threads -relativeimport -outdir
         ${CMAKE_CURRENT_BINARY_DIR}/osgeo ${SWIG_DEFINES} -I${PROJECT_SOURCE_DIR}/gdal -c++ -python -o ${ARRAY_OUTPUT}
         ${ARRAY_INPUT}
+      COMMAND ${CMAKE_COMMAND} ${WERROR_DEV_FLAG} "-DFILE=${CMAKE_CURRENT_BINARY_DIR}/osgeo/gdal_array.py" -P ${PROJECT_SOURCE_DIR}/swig/python/modify_cpp_files.cmake
       DEPENDS ${ARRAY_INPUT}
               ${PROJECT_SOURCE_DIR}/swig/include/python/typemaps_python.i
               ${CMAKE_CURRENT_SOURCE_DIR}/modify_cpp_files.cmake
@@ -107,13 +104,6 @@ set(GDAL_PYTHON_CSOURCES
         DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/osgeo/gdal_fsspec.py")
       list(APPEND GDAL_PYTHON_PYSOURCES "${CMAKE_CURRENT_BINARY_DIR}/osgeo/gdal_fsspec.py")
   endif()
-
-  foreach(_file IN ITEMS ${GDAL_PYTHON_CSOURCES})
-      add_custom_command(
-        OUTPUT ${_file}
-        APPEND
-        COMMAND ${CMAKE_COMMAND} ${WERROR_DEV_FLAG} "-DFILE=${_file}" -P ${CMAKE_CURRENT_SOURCE_DIR}/modify_cpp_files.cmake)
-  endforeach()
 
 add_custom_target(python_generated_files
   DEPENDS ${GDAL_PYTHON_PYSOURCES} ${GDAL_PYTHON_CSOURCES}

